fs-index/src/watch.rs

```diff
@@ -0,0 +1,69 @@
+use crate::{ResourceIndex, ARK_FOLDER};
+use anyhow::Result;
+use futures::{
+    channel::mpsc::{channel, Receiver},
+    SinkExt, StreamExt,
+};
+use log::info;
+use notify::{Config, Event, RecommendedWatcher, RecursiveMode, Watcher};
+use std::{fs, path::Path};
+
+/// Watch the root path for changes and update the index
+pub async fn watch_index<P: AsRef<Path>>(root_path: P) -> Result<()> {
+    log::debug!(
+        "Attempting to watch index at root path: {:?}",
+        root_path.as_ref()
+    );
+
+    let root_path = fs::canonicalize(root_path.as_ref())?;
+    let mut index = ResourceIndex::provide(&root_path)?;
+
+    let (mut watcher, mut rx) = async_watcher()?;
+    info!("Watching directory: {:?}", root_path);
+    let config = Config::default();
+    watcher.configure(config)?;
+    watcher.watch(root_path.as_ref(), RecursiveMode::Recursive)?;
+    info!("Started watcher with config: \n\t{:?}", config);
+
+    while let Some(res) = rx.next().await {
```

Comment:
From docs for `notify::event::Event`:
```
You might want to check Event::need_rescan to make sure no event was missed before you received this one.
```

This sounds like we should match `Event::need_rescan` and perform `update_all` in this case.

Reply:
Thanks. I didn't notice

Reply:
It seems that handling `Event::need_rescan` is still missing.